src/providers/dart_rename_provider.ts

```diff
@@ -0,0 +1,92 @@
+"use strict";
+
+import { window, workspace, RenameProvider, WorkspaceEdit, TextDocument, Position, CancellationToken, Uri, TextEdit, Range } from "vscode";
+import { Analyzer } from "../analysis/analyzer";
+import * as as from "../analysis/analysis_server_types";
+
+export class DartRenameProvider implements RenameProvider {	
+	private analyzer: Analyzer;
+	constructor(analyzer: Analyzer) {
+		this.analyzer = analyzer;
+	}
+	
+	provideRenameEdits(document: TextDocument, position: Position, newName: string, token: CancellationToken): Thenable<WorkspaceEdit> {		
+		return workspace.saveAll(false).then(() => {
```

Comment:
> the analysis service has all of the in-memory edits

Oh, I'm not aware of that. Yeah agree it would be unexpected, especially for unsaved files that's not affected by the rename. I've removed the `saveAll`, seems to work better this way.